libs/feature/catalog/src/lib/organisations/service/organisations-from-groups.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core'
+import { GroupApiModel, SearchApiService } from '@geonetwork-ui/data-access/gn4'
+import { ElasticsearchService, Organisation } from '@geonetwork-ui/util/shared'
+import { combineLatest } from 'rxjs'
+import { map, shareReplay } from 'rxjs/operators'
+import { GroupService } from '../../group/group.service'
+import { OrganisationsServiceInterface } from './organisations.service.interface'
+import { TranslateService } from '@ngx-translate/core'
+import { LANG_2_TO_3_MAPPER } from '@geonetwork-ui/util/i18n'
+
+const IMAGE_URL = '/geonetwork/images/harvesting/'
+
+type ESBucket = {
+  key: string
+  doc_count: number
+}
+interface OrganisationAggsBucket extends ESBucket {
+  mail: {
+    buckets: ESBucket[]
+  }
+}
+@Injectable()
+export class OrganisationsFromGroupsService
+  implements OrganisationsServiceInterface
+{
+  organisations$ = combineLatest([
+    this.fetchGroupsAggregation(),
+    this.groupService.groups$,
+  ]).pipe(
+    map(([groupsAgg, groups]) => {
+      return this.mapGroups(groupsAgg, groups)
+    }),
+    shareReplay()
+  )
+  organisationsCount$ = this.organisations$.pipe(map((orgs) => orgs.length))
+
+  constructor(
+    private esService: ElasticsearchService,
+    private searchApiService: SearchApiService,
+    private groupService: GroupService,
+    private translateService: TranslateService
+  ) {}
+
+  private fetchGroupsAggregation() {
```

Comment:
Just a question, wasn't it the purpose of the `FieldService` ?

Reply:
Not really, the `FieldService` is there to offer an abstraction on the search index and allow filtering and aggregating with agnostic field names.

The `OrganisationsFromGroupsService` also aggregates by group owner but this is an implementation detail, its only job is to output a list of organisations. I mean technically it does something very similar, but here we're not really doing anything related to _search_ in the sense of _searching records_. Another implementation could just query a `/api/groups` endpoint for example and that's it.

And speaking of which, the more I think about it the more I feel like the `feature-catalog` should not depend on the `feature-search` module at all; if anything, the dependency should be the other way around.